### PR TITLE
PROCESSINGS: User Access - Retrieve All User Accesses

### PR DESCRIPTION
**Pull request by cjdutoit**
closes #AB16819

### CODE REVIEW

File: ISL.ReIdentification.Core.Tests.Unit/Services/Processings/UserAccesses/UserAccessProcessingServiceTests.Exceptions.RetrieveAllUserAccesses.cs
Comment (JParkesAD) [inline ISL.ReIdentification.Core.Tests.Unit/Services/Processings/UserAccesses/UserAccessProcessingServiceTests.Exceptions.RetrieveAllUserAccesses.cs:1]: This file is for RetrieveAllUserAccesses but the methods mocked and used throughtout are the AddUserAccess

File: ISL.ReIdentification.Core/Services/Processings/UserAccesses/UserAccessProcessingService.Exceptions.cs [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        private async ValueTask<IQueryable<UserAccess>> TryCatch(ReturningUserAccessesFunction
+            returningUserAccessesFunction)
+        {
+            try
+            {
+                return await returningUserAccessesFunction();
+            }
+            catch (UserAccessValidationException userAccessValidationException)
+            {
+                throw await CreateAndLogDependencyValidationExceptionAsync(userAccessValidationException);
```
Comment (JParkesAD) [inline ISL.ReIdentification.Core/Services/Processings/UserAccesses/UserAccessProcessingService.Exceptions.cs:66]: Should this use CreateAndLogValidationExceptionAsync instead

Comment (JParkesAD) [reply ISL.ReIdentification.Core/Services/Processings/UserAccesses/UserAccessProcessingService.Exceptions.cs:66]: Or do we need this catch at all as we are not validating anything on retrieve all?

Comment (cjdutoit) [reply ISL.ReIdentification.Core/Services/Processings/UserAccesses/UserAccessProcessingService.Exceptions.cs:66]: > Should this use CreateAndLogValidationExceptionAsync instead

@JParkesAD  This looks correct.  UserAccessValidationException is the validation error from the dependency.

This is a processing service, so it uses these for the normal validation on this level that will be wrapped as `UserAccessProcessingValidationException`:  
```cs
            catch (NullUserAccessProcessingException nullUserAccessProcessingException)
            {
                throw await CreateAndLogValidationExceptionAsync(nullUserAccessProcessingException);
            }
            catch (InvalidUserAccessProcessingException invalidUserAccessProcessingException)
            {
                throw await CreateAndLogValidationExceptionAsync(invalidUserAccessProcessingException);
            }
```

**Pull request merged**